cmake-edit-core/src/main/java/ru/urururu/cmakeedit/core/checker/SimulationState.java

```diff
@@ -12,19 +12,109 @@
+        StringBuilder sb = new StringBuilder();
+
+        argumentNode.visitAll(new NodeVisitorAdapter() {
+            @Override
+            public void accept(ExpressionNode node) {
+                StringBuilder exprBuilder = new StringBuilder();
+
+                if (node.getKey() != null) {
+                    exprBuilder.append(node.getKey());
+                }
+
+                node.getNested().stream().forEach(n -> exprBuilder.append(getNodeFromState(n)));
+
+                String expr = exprBuilder.toString();
+                getValue(expr);
+
+                sb.append('*').append(expr);
+            }
+
+            @Override
+            public void accept(ConstantNode node) {
+                sb.append(node.getValue());
+            }
+        });
+
+        return sb.toString();
+    }
+
+    String getValue(ArgumentNode argumentNode) {
+        getNodeFromState(argumentNode);
+
+        return getValue(getArgument(argumentNode));
+    }
+
+    String getValue(String expression) {
+        int expressionStart = expression.indexOf("${"); // todo add support for ENV
+        int expressionEnd = expression.lastIndexOf('}');
+
+        String result;
+        if (expressionStart != -1 && expressionEnd > expressionStart) {
+            String sub = expression.substring(expressionStart + 2, expressionEnd);
+
+            // todo remove suspicious point for sub or for getValue(sub)?
+
+            result = expression.substring(0, expressionStart) + getValue(sub) + expression.substring(expressionEnd);
+        } else {
+            result = expression;
+        }
+
+        suspiciousPoints.removeAll(variables.getOrDefault(result, Collections.emptySet()));
+
+        return result;
+    }
+
+    void putValue(ArgumentNode argumentNode, CommandInvocationNode command) {
+        putValue(argumentNode, command, false);
+    }
+
+    void putValue(ArgumentNode argumentNode, CommandInvocationNode command, boolean parentScope) {
+        // expressions from argumentNode should be used
+
+        // todo inline argumentNode via getValue()
+        variables.put(getNodeFromState(argumentNode), Collections.singleton(command));
+    }
+
+    static String getArgument(Node argumentNode) {
+        StringBuilder sb = new StringBuilder();
+
+        argumentNode.visitAll(new NodeVisitorAdapter() {
+            @Override
+            public void accept(ExpressionNode node) {
+                if (node.getKey() != null) {
+                    sb.append(node.getKey());
+                }
+                //throw new IllegalStateException("not implemented");
+                node.getNested().stream().forEach(n -> n.visitAll(this));
+            }
+
+            @Override
+            public void accept(ConstantNode node) {
+                sb.append(node.getValue());
+            }
+        });
+
+        return sb.toString();
+    }
+
+    void simulate(Set<CommandInvocationNode> suspiciousPoints, Node node) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused method parameter "suspiciousPoints". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1172)